exchanges/binance/binance.go

```diff
@@ -161,27 +162,130 @@ func (b *Binance) GetHistoricalTrades(symbol string, limit int, fromID int64) ([
 	return nil, common.ErrFunctionNotSupported
 }
 
-// GetAggregatedTrades returns aggregated trade activity
-//
-// symbol: string of currency pair
-// limit: Optional. Default 500; max 1000.
-func (b *Binance) GetAggregatedTrades(symbol string, limit int) ([]AggregatedTrade, error) {
-	var resp []AggregatedTrade
-
-	if err := b.CheckLimit(limit); err != nil {
-		return resp, err
+// GetAggregatedTrades returns aggregated trade activity.
+// If more than one hour of data is requested or asked limit is not supported by exchange
+// then the trades are collected with multiple backend requests.
+// https://binance-docs.github.io/apidocs/spot/en/#compressed-aggregate-trades-list
+func (b *Binance) GetAggregatedTrades(arg *AggregatedTradeRequestParams) ([]AggregatedTrade, error) {
+	params := url.Values{}
+	params.Set("symbol", arg.Symbol)
+	// fromId xor start time must be set
+	canBatch := arg.FromID == 0 != arg.StartTime.IsZero()
+	needBatch := false
+	if arg.Limit > 0 {
+		err := b.CheckLimit(arg.Limit)
```

Comment:
Last last last last change from me promise lol. but if valid limits dont really apply to this function. so can we just remove the arg limit check n instead just add one for if > 1000. coz the endpoint doesnt restrict requests to 5, 10, 15, ..., 500, 1000. we can put in 69 or 111 or any random number below 1000 and theyll return the exact amt

Reply:
aha, great catch! The CheckLimit was there before so I never questioned whether this was necessary. Removed.